FenwickTree/RangeUpdatePointQueries.java

```diff
@@ -0,0 +1,87 @@
+public class RangeUpdatePointQueries {
```

Comment:
It'd be great if you could provide tests for your Fenwick tree implementation.

Reply:
I will add some tests based on tests you have for any other algorithms. I will push with all the changes below in a day.